sklearn/ensemble/gradient_boosting.py

```diff
@@ -195,16 +195,23 @@ def update_terminal_regions(self, tree, X, y, residual, y_pred,
         ----------
         tree : tree.Tree
             The tree object.
-        X : np.ndarray, shape=(n, m)
+        X : ndarray, shape=(n, m)
             The data array.
-        y : np.ndarray, shape=(n,)
+        y : ndarray, shape=(n,)
             The target labels.
-        residual : np.ndarray, shape=(n,)
+        residual : ndarray, shape=(n,)
             The residuals (usually the negative gradient).
-        y_pred : np.ndarray, shape=(n,):
+        y_pred : ndarray, shape=(n,):
             The predictions.
-        sample_weight : np.ndarray, shape=(n,):
+        sample_weight : ndarray, shape=(n,):
             The weight of each sample.
+        sample_mask : ndarray
+            The sample mask to be used.
+        learning_rate : float, default=0.1
+            learning rate shrinks the contribution of each tree by 
+            `learning_rate`.
+        k : int, default 0
```

Comment:
It looks like `y_pred` should have shape `(n, n_estimators)` and `k` is the index of the estimator being updated.